src/abstractdatatable/reshape.jl

```diff
@@ -193,30 +188,17 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
     # `rowkey` integer indicating which column to place along rows
     # `colkey` integer indicating which column to place along column headers
     # `value` integer indicating which column has values
-    refkeycol = NullableCategoricalArray(dt[rowkey])
```

Comment:
This one is interesting. I figured out why unstack and stack have the ability to introduce missing data and it's precisely because of the use of (Nullable)CategoricalArrays here and in unstack. As shown [here](https://github.com/JuliaData/DataTables.jl/pull/30/files#diff-adff67e35c47c918a01474bd48196953R289) (and the few lines below) stack and unstack will actually add data into the table that does not exist in the table prior to stack/unstack because it uses the levels of the CategoricalArrays rather than the data present in the table. So with these changes 1. we don't have to do any array conversion, presumably saving the cost of multiple array conversions, and 2. the results will depend on the data present in the table rather than what may or may not be present in the levels of the CategoricalArray. If the behavior of inserting missing data for missing levels is more desirable than retaining column type and avoiding array conversions, then I'm happy to flip this back. My thought here was that if users want to account for missing levels, why did they remove them from their table in the first place? Leave the nulls in the table before calling stack/unstack....